### PR TITLE
simplified pickle interface to rely on base recording/sorting extractors

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@alejoe91 Significant reduction in duplicated code for running the interface. I agree that long-term we should look into mix-ins for separating the internal class configuration (to adhere better to inheritance) from the common set of conversion functionality, but that will likely take a bit of debugging to streamline.

### CODE REVIEW

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py [hunk 1]
```diff
@@ -21,51 +19,16 @@ def get_input_schema(cls):
         )
 
     def __init__(self, **input_args):
-        super().__init__(**input_args)
+        self.input_args = input_args
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:22]: why is this line needed?

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py [hunk 2]
```diff
@@ -74,45 +37,5 @@ def get_input_schema(cls):
...
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:40]: why is this line needed?

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:22]: Previously, when this inherited from BaseDataInterface, calling super().__init__ calls self.input_args = input_args.

The other alternative here is to call super().super().__init__ but that seems a little convoluted (let me know if you'd prefer it, though).

Having the record of received input arguments is very useful for both debugging and retrieving information for stuff like metadata.

Comment (bendichter) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:24]: why do we need input_args to be assigned as an attribute of the class?

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-    get_schema_from_hdmf_class
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ..basesortingextractorinterface import BaseSortingExtractorInterface
 
 
-class SIPickleRecordingExtractorInterface(BaseDataInterface):
+class SIPickleRecordingExtractorInterface(BaseRecordingExtractorInterface):
+    """Primary interface for reading and converting SpikeInterface objects through Pickle files."""
+
+    RX = None
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:11]: Why is this needed?

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:11]: With this version that inherits from the BaseRecordingExtractorInterface, I assume it would only be proper to contain the RX class object common to all RecordingExtractorInterfaces, with the None value indicating it is not by default any particular class from SpikeExtractor.

That said, since we're not relying on the class for instantiation of the actual instance of the extractor in the __init__, I suppose it wouldn't actually get used unless something downstream wanted to test whether this interface has an RX object that is None or something.

We could also update the status of RX perhaps after loading the recording object during the __init__ here, I think.

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:24]: All inheritors of the [base data interfaces](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/basedatainterface.py) have this. Downstream stuff may or may not wish to access it as a shortcut for retrieving file paths and stuff of that nature

Comment (bendichter) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:24]: ok sure

File: nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py (hunk 3 above)
Comment (bendichter) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:11]: This class would already inherit the `RX = None` from the `BaseRecordingExtractorInterface` so this line is redundant

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/sipickledatainterfaces.py:11]: Oh, that's true. Fixing that now
